Content.Server/Chemistry/Components/ChemMasterComponent.cs

```diff
@@ -23,5 +25,22 @@ public sealed partial class ChemMasterComponent : Component
 
         [DataField("clickSound"), ViewVariables(VVAccess.ReadWrite)]
         public SoundSpecifier ClickSound = new SoundPathSpecifier("/Audio/Machines/machine_switch.ogg");
+
+        // Frontier: upgrade fields
+        [DataField]
+        public uint[] PillDosageLimitPerTier;
+
+        [DataField]
+        public ProtoId<MachinePartPrototype> PillDosageMachinePart = "Manipulator";
+
+        [ViewVariables]
+        public uint MaxVolume;
```

Comment:
**Question** (non-blocker): by storing the max volume here _and_ on the solution, is there any risk of them divering? I.e., are there any places where we might forget to update one of them? Might it be possible to rely entirely on the solution max volume? Maybe that's not available everywhere we need it.